lib/Runtime/Library/JavascriptBigInt.cpp

```diff
@@ -71,21 +91,44 @@ namespace Js
         return true;
     }
 
+    bool JavascriptBigInt::Resize(digit_t length)
+    {
+        digit_t *digits;
+
+        if (length <= m_maxLength)
+            return true;
+
+        length += length;//double size
+        if (m_digits == m_reservedDigitsInit)
+        {
+            if ((SIZE_MAX / sizeof(digit_t) < length) || (NULL == (digits = (digit_t *)malloc(length * sizeof(digit_t)))))
```

Comment:
We should RecyclerNewArrayLeafZ rather than malloc/realloc. Then the recycler can manage cleaning up the memory without having to do complicated finalization logic for this class. "Leaf" means the allocation doesn't contain any pointers to other recycler objects, so the recycler doesn't waste time scanning it or risk misinterpreting some plain data as a pointer and keeping stuff alive when it shouldn't.

Also, I'd prefer to have the allocation and assignment in its own statement rather than hidden out halfway through a long conditional.

Reply:
fixed